charmcraft/commands/store/__init__.py

```diff
@@ -722,6 +737,222 @@ def run(self, parsed_args):
         emit.message(msg.format(*args))
 
 
+class PromoteBundleCommand(BaseCommand):
+    """Promote a bundle in the Store."""
+
+    name = "promote-bundle"
+    needs_config = True
```

Comment:
No longer use `needs_config`.

Reply:
`needs_config` isn't used directly, but it overrides the `needs_config` in `BaseCommand`. Commands with `needs_config = False` don't have to be run with a project directory specified (or in a project directory).